refs #138: Aseprite importer for inventory items

stickgrinder opened the pull request:
This provides:

* Main functionality
* A migration for the inventory items that adds an AnimationPlayer
* General polishing to the importers code

The way we save tags options in PopochiuResources must be carefully considered with @mapedorr .

stickgrinder commented:
@mapedorr this is ready for review. I know it's huge (sorry), but this was the last "elephant in the room" about importers.

I still want to address a couple of UX improvements, but it's trivial stuff, that I can close during the next weekend for sure (everything tracked in other issues!) so, if you manage to review this work of mine, we can move forward on other parts of the plugin :)

Yay

mapedorr commented on addons/popochiu/editor/importers/aseprite/docks/aseprite_importer_dock.gd:
```diff
@@ -359,7 +363,10 @@ func _on_tags_title_toggled(button_pressed: bool) -> void:
 func _set_tags_visible(is_visible: bool) -> void:
-	get_node("%Tags").visible = is_visible
+	# If the tags container is empty, we show an info box
+	get_node("%TagsInfo").visible = (get_node("%Tags").get_child_count() == 0)
+
+	get_node("%TagsScrollContainer").visible = is_visible
```
Are these uniquely named nodes? Because if they are, you could do:

```suggestion
	%TagsInfo.visible = %Tags.get_child_count() == 0

	%TagsScrollContainer.visible = is_visible
```

Otherwise, ignore this comment.

mapedorr commented on addons/popochiu/editor/importers/aseprite/docks/aseprite_importer_dock_inventory.gd:
```diff
@@ -0,0 +1,139 @@
+## Check if an inventory item with the given name already exists and load it.
+## Returns the loaded inventory item scene or null if it doesn't exist.
+func _get_existing_inventory_item(item_name: String) -> PopochiuInventoryItem:
+	var item_path := PopochiuResources.INVENTORY_ITEMS_PATH.path_join(
+		"%s/inventory_item_%s.tscn" % [item_name.to_snake_case(), item_name.to_snake_case()]
+	)
+	
+	if not FileAccess.file_exists(item_path):
+		return null
+	
+	var packed_scene: PackedScene = load(item_path)
+	if not packed_scene:
+		return null
```
You could do something similar asking for the inventory item to the `popochiu_data.cfg`:

```gdscript
var item_res_path: String = PopochiuResources.get_data_value("inventory_items", item_name.to_snake_case(), null)
if not item_res_path:
	return null

var packed_scene: PackedScene = load(load(item_res_path).scene)
```

mapedorr commented on addons/popochiu/migration/migrations/popochiu_migration_6.gd:
```diff
@@ -0,0 +1,97 @@
+	
+	while file_name != "":
+		if dir.current_is_dir():
+			var item_scene_path := inventory_path.path_join(file_name).path_join("inventory_item_%s.tscn" % file_name)
+			if FileAccess.file_exists(item_scene_path):
```
As I mentioned in a previous comment, this could also be done by accessing the `popochiu_data.cfg` file.

stickgrinder replied:
Done

mapedorr reviewed (changes requested):
Looks solid, but since I know you still want to polish a couple of things, I requested some changes and made some suggestions before approving the PR.

Tomorrow morning I'll try to make some tests in order to find bugs one can't see only by looking at the code.

stickgrinder commented on addons/popochiu/editor/importers/aseprite/animation_creator.gd:
Done!

stickgrinder commented on addons/popochiu/editor/importers/aseprite/animation_creator.gd:
```diff
@@ -301,52 +354,87 @@ func _remove_properties_from_path(path: NodePath) -> NodePath:
-	animation.track_insert_key(hframes_track_index, 0, _target_sprite.hframes)
+	# Only create hframes/vframes tracks for Sprite2D nodes
+	if _target_sprite is Sprite2D:
+		var hframes_track = _get_property_track_path("hframes")
+		var hframes_track_index = _create_track(_target_sprite, animation, hframes_track)
```
Done

stickgrinder commented:
@mapedorr I did some changes to this PR:

1. Addressed all your comments on the code
2. Split the animation_creator.gd script and based it on the strategy pattern, so we have specialized classes for recipient nodes (Sprite2D, TextureRect) allowing us to support more in the future.
3. Hardened some other small parts where I saw the opportunity

I mentioned in my previous comment that I wanted to work on some UX improvement, but I splitted them in #403, so I would prefer to have this merged as soon as you test it.

I'll solve #403 afterwards, add all the necessary documentation, and **finally complete the aseprite importer epic.**

Thanks

stickgrinder commented:
Merging with permission from @mapedorr
